src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -207,6 +207,16 @@ internal virtual ModuleSymbol ContainingModule
             }
         }
 
+        /// <summary>
+        /// The index of this member in the containing symbol. This is an optional
+        /// property, implemented by anonymous type properties only, for comparing
+        /// symbols in flow analysis. Returns -1 if not implemented.
+        /// </summary>
+        /// <remarks>
+        /// Should this be used for tuple fields as well?
+        /// </remarks>
+        internal virtual int MemberIndex => -1;
```

Comment:
Rather than -1 as  magic number did you consider `int?`?

Reply:
Yes, but the caller then needs to use `HasValue` and `GetValueOrDefault()`.

---
In reply to: [259080867](https://github.com/dotnet/roslyn/pull/33562#discussion_r259080867) [](ancestors = 259080867)

Reply:
Hard to avoid that. Easy to avoid the `-1` check 😄